### PR TITLE
Angular tests

### PR DESCRIPTION
**Pull request by Raxerz**
This contains the tests for angular components.

### CODE REVIEW

File: src/tests/Authentication/login.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        component = fixture.componentInstance;
+        authService = fixture.debugElement.injector.get(AuthService);
+        apiService = TestBed.get(APIService);             
+    });
+
+    it('should be created', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should route to menu page on successful login', () => {
```
Comment (yukie1412) [inline src/tests/Authentication/login.component.spec.ts]: spacing

Comment (Raxerz) [inline src/tests/Authentication/login.component.spec.ts]: Where exactly is the spacing issue @yukie1412

**Comment by Raxerz** [comment]
I have covered all the core methods of all components and services @yukie1412.
There are about 50 test cases.

File: src/tests/Authentication/login.component.spec.ts (hunk 1 above)
Comment (Raxerz) [inline src/tests/Authentication/login.component.spec.ts]: Is it fine now @yukie1412 ?

File: src/tests/Services/auth.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            providers: [
+                AuthService, 
+                APIService,
+                {
+                  provide: XHRBackend,
+                  useClass: MockBackend
+                }               
+            ],
+			imports: [
+				RouterTestingModule,
```
Comment (yukie1412) [inline src/tests/Services/auth.service.spec.ts]: spacing

File: src/tests/Services/dashboard.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    beforeEach(() => {
+            TestBed.configureTestingModule({
+            imports: [
+                HttpModule
+            ], 
+            providers: [
+                DashboardService,
+                APIService,
+                {
+                  provide: XHRBackend,
```
Comment (yukie1412) [inline src/tests/Services/dashboard.service.spec.ts]: spacing

File: src/tests/Services/navbar.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+const getUsername = baseAPIUrl + 'username';
+
+describe('NavbarService', () => {
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            providers: [
+                NavbarService, 
+                APIService,
+                {
+                  provide: XHRBackend,
```
Comment (yukie1412) [inline src/tests/Services/navbar.service.spec.ts]: spacing

**Comment by Raxerz** [comment]
Made the changes @yukie1412 . Please check.

File: src/tests/Services/auth.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            })
+        )
+    });    
+
+    it('should be created', inject([AuthService], (service: AuthService) => {
+        expect(service).toBeTruthy();
+    }));
+
+    it('should clear token from local storage on logging out', inject([AuthService], (service: AuthService) => {
+                service.logout()
```
Comment (yukie1412) [inline src/tests/Services/auth.service.spec.ts]: spacing

File: src/tests/Services/auth.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                        .subscribe(res => {
+                            expect(localStorage.getItem(localStorageKey)).toBe(null);
+                        });
+    }));
+    it('should add token to localstorage on logging in', inject([AuthService], (service: AuthService) => {               
+                service.loginUser({email: 'abc@gmail.com', password: 'abc'})
```
Comment (yukie1412) [inline src/tests/Services/auth.service.spec.ts]: spacing

File: src/tests/Services/auth.service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+                        .subscribe(res => {
+                            expect(res).toBeTruthy();
+                            expect(localStorage.getItem(localStorageKey)).toBeDefined();
+                        });
+    })); 
+    it('should check for authentication', inject([AuthService], (service: AuthService) => {               
+                service.authenticated()
```
Comment (yukie1412) [inline src/tests/Services/auth.service.spec.ts]: spacing

File: src/tests/Services/dashboard.service.spec.ts [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+
+const mockUserInfoResponse = {user: {email: 'abc@gmail.com', name: 'Rajath'}};
+
+const mockMailPCPolicyResponse = [];
+mockMailPCPolicyResponse.push({message: 'Mail Sent Succesfully.'});
+mockMailPCPolicyResponse.push({error: 'Something Went Wrong! Try again later.'});
+
+describe('DashboardService', () => {
+    beforeEach(() => {
+            TestBed.configureTestingModule({
```
Comment (yukie1412) [inline src/tests/Services/dashboard.service.spec.ts]: spacing

File: src/tests/Services/dashboard.service.spec.ts [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+            })
+        )
+    });    
+
+    it('should be created', inject([DashboardService], (service: DashboardService) => {
+        expect(service).toBeTruthy();
+    }));
+
+    it('should get user info', inject([DashboardService], (service: DashboardService) => {
+                service.getUserInfo()
```
Comment (yukie1412) [inline src/tests/Services/dashboard.service.spec.ts]: spacing

File: src/tests/Services/dashboard.service.spec.ts [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+                        .subscribe(res => {
+                            expect(res).toBe(mockUserInfoResponse);
+                        });
+    }));
+
+    it('should get user progress status', inject([DashboardService], (service: DashboardService) => {
+                service.getProgressStatus()
```
Comment (yukie1412) [inline src/tests/Services/dashboard.service.spec.ts]: spacing, & all functions below

File: src/tests/Services/navbar.service.spec.ts [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+            })
+        )
+    });    
+
+    it('should be created', inject([NavbarService], (service: NavbarService) => {
+        expect(service).toBeTruthy();
+    }));
+
+    it('should return the username', inject([NavbarService], (service: NavbarService) => {
+                service.getUserName()
```
Comment (yukie1412) [inline src/tests/Services/navbar.service.spec.ts]: spacing

**Pull request merged**